### DISCUSSION TITLE
Welcome to svgedit Discussions!

### DISCUSSION THREAD

**Discussion by jfhenon** [SVG-Edit/svgedit#452] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by warrenvail** [comment]
My name is Warren Vail, I am a developer and I am developing a web application that needs a web based drawing pluggin to allow people (mostly pilots) who are building their own airplane, but need to be able to manage their construction project and share drawings of what they are doing or considering doing in an accurate way to get good advice from others to make good decisions.  Technically the application could be used to handle any project like that (airplanes, boats, cars, etc) and but to provide plans and construction details for that many projects with drawings that cover all the details, I need to have a means of loading and revising the drawings contained in the plans used for all the projects, and the ideal way to do that is to include a editor for drawings that takes applies the best technologies to people who are focused on their projects and not piling up charges that take away from their investment in their own safety where it counts most, the parts of their project. To do this I need to be able to draw the parts of a airplane, in my case I have a set of plans that I can scan drawings from existing plans (no longer protected by copyright), and trace those scans in a detail editor, but produce the more scalable drawings with SVG, and present a vanilla set of plans without the pencil marks of previous builders, with corrected drawings where that is appropriate, and other data.  I am trying to embed the whole SVG-Editor (and finding difficulty), and lend back to the tool from my own experience with my project, for example not all rectangles are parallel or perpendicular to the bottom of the page for very valid reasons, yet the sides of the rectangle are parallel to each other and all corners are right angles. If I find solutions to these problems I would love to see this open source project benefit (My project will be open source as well).  If I am successful, perhaps some day this tool will be used for space craft (Me, I am building a BD-5 Turbo Prop Version of the James Bond Jet).

**Reply by jfhenon** [reply]
Hello Warren, thank your for your introduction and indeed your project is really interesting. I hope you can use SVGEdit to achieve your requirements. As you may have seen in other discussions, we are in the middle of a quite significant refactoring. I hope to see a working Beta by January. For now, you may use V5 or V6 but it's even better if you want to contribute to V7.

**Reply by warrenvail** [reply]
I ran into problems trying to embed V6 into my module, so I am backing away for a while, will look for the new release in Jan/Feb? and try again.  I really do believe there is a need for a CAD tool that supports engineers who want to collaborate with each other.  As I try to retire on a budget, I am very much a supporter of Open Source options without trying to turn them into a profit item, and hope I get the chance to add some good options to your tool.

**Comment by LECUYER-David** [comment]
Hello

I will need your help for a useful little improvement.
1 / I work local storage
2 / when I close the window, this is when the information is stored in the local storage to retrieve its image the next time it is opened.
3 / buggs or inconvenience.
I have 3PC at home (google, firefox).
everything works flawlessly when the window is closed for saving to local storage, except on 1PC with firefox it does not work.
PS (I have checked everything options, version, etc ...) and I do not know the reason.
suddenly every time everything is lost.
4 / idea or proposal
is it possible when we choose the local storage, to make a regular backup, for example every 3 seconds for example (I suppose with interval in js)
-this will avoid the doubt of operation of certain browser when closing the window.
-Other possible case: if someone works for 1/2 hour or 1 hour on the svg-edit, and there is a power cut, at that moment nothing has been saved in their PC.

Please inform me if it is possible to improve this operation.

because I find the work that has been done that svg-edit is awesome.

cordially

**Reply by jfhenon** [reply]
Hello David, I think it's not too difficult to do your suggestion and it would indeed be a good improvement. I think we could define a delay or a number of changes before saving to the localstorage. You could add it in the issue tab so we keep it in our radar. Merci :-)

**Reply by LECUYER-David** [reply]
Hello jfhenon,

Thank you for your superb idea of ​​saving following a number of actions carried out.
this is exactly what i need.




Thanks again for your help

cordially
David LECUYER
My mail :davlecuyer@yahoo.fr

**Comment by nfjdu** [comment]
Hi! Is there any convenient opportunity to resize elements through the code? I investigate your codebase and seems the `recalculate.js` includes some code responsible, as I understand, for applying the changes, somehow collected earlier. How could I, for example, prepare `changes` data and then use `recalculateDimensions` (or some another func if I misunderstood this func sense) to apply them? Thank you so much

**Reply by LECUYER-David** [reply]
Hello,

I don't know if this answer can help you, but on my side I resize against the localstorage data
example:
1/your site you want a width of 370px.
2/I read the user's local storage
`<svg width="1024" height="768" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">`

3/height recalculation=1024x370/768=493
4/add viewbox with original sizes
`<svg width="1024" height="768" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 1024 768" >`

5/replace height and width with the sizes of your site
`<svg width="370" height="493" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 1024 768" >`

6/display svg 

I hope this can help you with your calculate Dimensions or not (otherwise sorry for having answered your question incorrectly).

cordially

David LECUYER

**Reply by nfjdu** [reply]
Unfortunately, seems like it's not what I actually need, but anyway thank you for the answer!

**Reply by nfjdu** [reply]
Seems I was wrong..`cords.js` and its `remapElement` function is responsible for the actions I need
